File: src/main/resources/scripts/fib.rb
# Iterative Implementation
def fib(n)
  return 0 if n.zero?

  prv, f = 1 , 1
  t=n.abs - 2
  t.times { prv, f = f, f + prv }
  
  if n < 0
    n+=1
    f*=-1**n 
  else
    f*=1
  end

end

(1..11).each{|i|print "#{fib(i)} "}
puts

File: src/main/resources/scripts/fib.py
# Analytic Implementation
from math import *
 
def fib(n):
  s = sqrt(5);
  p = (1 + s) / 2;
  q = 1 / p;
  t = p**n + q**n;
  t = t / s;
  return int( t + 0.5 )
 
for i in range(1,12):
  print fib(i),

print ""